src/client/clientmap.cpp

```diff
@@ -31,6 +31,37 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include <algorithm>
 #include "client/renderingengine.h"
 
+// struct MeshBufListList
+void MeshBufListList::clear()
+{
+	for (auto &list : lists)
+		list.clear();
+}
+
+void MeshBufListList::add(scene::IMeshBuffer *buf, v3s16 position, u8 layer)
+{
+	// Append to the correct layer
+	std::vector<MeshBufList> &list = lists[layer];
+	const video::SMaterial &m = buf->getMaterial();
+	for (MeshBufList &l : list) {
+		// comparing a full material is quite expensive so we don't do it if
+		// not even first texture is equal
+		if (l.m.TextureLayer[0].Texture != m.TextureLayer[0].Texture)
+			continue;
+
+		if (l.m == m) {
+			l.bufs.push_back({position,buf});
```

Comment:
if you're moving this code anyway:
```suggestion
			l.bufs.emplace_back({position, buf});
```